crates/cli/src/commands/patterns_test.rs

```diff
@@ -276,7 +297,232 @@ pub(crate) async fn run_patterns_test(
         bail!("No testable patterns found. To test a pattern, make sure it is defined in .grit/grit.yaml or a .md file in your .grit/patterns directory.");
     }
     info!("Found {} testable patterns.", testable_patterns.len(),);
-    get_marzano_pattern_test_results(testable_patterns, &libs, arg, flags.into()).await
+    get_marzano_pattern_test_results(testable_patterns, &libs, &arg, flags.clone().into()).await?;
+
+    if arg.watch {
+        enable_watch_mode(arg, flags);
+    }
+    Ok(())
+}
+
+fn enable_watch_mode(cmd_arg: PatternsTestArgs, cmd_flags: GlobalFormatFlags) -> () {
+    let path = Path::new(".grit");
```

Comment:
We don't want to watch all of `.grit` (including .gritmodules). Also this should probably not be created new, but instead should be passed in based on the original patterns retrieved.

Reply:
notify crate currently does not support ignoring the directories inside tracked directory(https://github.com/notify-rs/notify/issues/598). I have added a temporary fix to avoid re-running the tests in case any of these files are modified.

Reply:
> should be passed in based on the original patterns retrieved
let's say a new pattern is added to a file, we will not know that unless we re-read the patterns from file scratch. right?

Reply:
> let's say a new pattern is added to a file, we will not know that unless we re-read the patterns from file. right?

That's true, I wasn't thinking of the case of an entirely new file. This is ok then.